pkg/utils/k8sutil.go

```diff
@@ -865,16 +899,29 @@ func EnsureFuncCronJob(client kubernetes.Interface, funcObj *spec.Function, or [
 		},
 	}
 
-	_, err := client.BatchV2alpha1().CronJobs(funcObj.Metadata.Namespace).Create(job)
+	// We need to use directly the REST API since the endpoint
+	// for CronJobs changes from Kubernetes 1.8
+	jobJSON, err := json.Marshal(job)
+	if err != nil {
+		return err
+	}
+	_, err = doRESTCronJobReq(client, groupVersion, "create", jobName, funcObj.Metadata.Namespace, &jobJSON)
 	if err != nil && k8sErrors.IsAlreadyExists(err) {
-		var data []byte
-		data, err = json.Marshal(job)
+		newCronJob := batchv2alpha1.CronJob{}
+		newCronJob, err = doRESTCronJobReq(client, groupVersion, "get", jobName, funcObj.Metadata.Namespace, nil)
 		if err != nil {
 			return err
 		}
-		_, err = client.BatchV2alpha1().CronJobs(funcObj.Metadata.Namespace).Patch(job.Name, types.StrategicMergePatchType, data)
+		newCronJob.ObjectMeta.Labels = funcObj.Metadata.Labels
+		newCronJob.ObjectMeta.OwnerReferences = or
+		newCronJob.Spec = job.Spec
+		newCronJobJSON := []byte{}
```

Comment:
make it simple:

```newCronJobJSON, err := json.Marshal(newCronJob)```

Reply:
It is actually intentional. This is to avoid to define an object `err` inside the `if` scope. If we don't do the return at the end of the `if` the value of `err` outside won't be the value set inside. I can change it to do the return at the end if that is easier to understand.